### DISCUSSION TITLE
Is it okay for the token to be a unique database field?

### DISCUSSION THREAD

**Discussion by seejamescode** [nextauthjs/next-auth#4191] [Help]
I am currently using `generateVerificationToken` to shorten the token to a six digit login code in case the user wants to copy and paste the token into the app instead of a magic link. I'm using Prisma and noticed the schema has `token` as the unique identifier for the `VerificationToken` model.

Would that give a 1/10000 chance that a token is a duplicate and be invalid to the schema? If so, should we make a different unique identifier for `VerificationToken` that isn't the token used to login?

**Comment by youminkim** [comment]
Same question here. We use magic code approach with 4 digit codes and the chance is pretty high. 

https://www.ramielcreations.com/nexth-auth-magic-code
